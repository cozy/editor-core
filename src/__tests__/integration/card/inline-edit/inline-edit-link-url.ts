import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import {
  getDocFromElement,
  editable,
  copyToClipboard,
  gotoEditor,
  linkUrlSelector,
} from '../../_helpers';

import { waitForInlineCardSelection } from '@atlaskit/media-integration-test-helpers';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';

BrowserTestCase(
  'card: changing the link URL of an inline link to another supported link should reresolve smart card',
  { skip: ['safari', 'edge', 'firefox'] },
  async (client: ConstructorParameters<typeof Page>[0], testName: string) => {
    const page = new Page(client);

    // Copy stuff to clipboard
    await copyToClipboard(page, 'https://www.atlassian.com');

    await gotoEditor(page);

    // Paste the link
    await page.paste();
    await page.keys('Enter');

    await waitForInlineCardSelection(page);

    await page.click('button[aria-label="Edit link"]');
    // Clear the Link Url field before typing
    await page.clear(linkUrlSelector);

    // Change the 'link address' field to another supported link and press enter
    await page.type(linkUrlSelector, [
      'https://product-fabric.atlassian.net/wiki/spaces/E/overview',
    ]);
    await page.keys('Enter');

    // Ensure smart card still exists
    await waitForInlineCardSelection(page);
    expect(
      await page.$eval(editable, getDocFromElement),
    ).toMatchCustomDocSnapshot(testName);
  },
);
