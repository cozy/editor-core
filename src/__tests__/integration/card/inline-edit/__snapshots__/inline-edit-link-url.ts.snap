// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card: changing the link URL of an inline link to another supported link should reresolve smart card 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "attrs": Object {
            "data": Object {
              "@context": "https://www.w3.org/ns/activitystreams",
              "@type": "Document",
              "name": "https://product-fabric.atlassian.net/wiki/spaces/E/overview",
              "url": "https://product-fabric.atlassian.net/wiki/spaces/E/overview",
            },
            "url": null,
          },
          "type": "inlineCard",
        },
        Object {
          "text": "  ",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
    Object {
      "type": "paragraph",
    },
  ],
  "type": "doc",
}
`;
